src/editors/containers/ProblemEditor/data/OLXParser.js

```diff
@@ -91,18 +92,74 @@ export class OLXParser {
       },
       processEntities: false,
     };
+    const richTextBuilderOptions = {
+      ignoreAttributes: false,
+      numberParseOptions: {
+        leadingZeros: false,
+        hex: false,
+      },
+      preserveOrder: true,
+      processEntities: false,
+    };
     // There are two versions of the parsed XLM because the question requires the order of the
     // parsed data to be preserved. However, all the other widgets need the data grouped by
     // the wrapping tag.
-    const questionParser = new XMLParser(questionOptions);
+    const richTextParser = new XMLParser(richTextOptions);
     const parser = new XMLParser(parserOptions);
     this.builder = new XMLBuilder(builderOptions);
+    this.richTextBuilder = new XMLBuilder(richTextBuilderOptions);
     this.parsedOLX = parser.parse(olxString);
-    this.parsedQuestionOLX = questionParser.parse(olxString);
+    this.richTextOLX = richTextParser.parse(olxString);
     if (_.has(this.parsedOLX, 'problem')) {
       this.problem = this.parsedOLX.problem;
-      this.questionData = this.parsedQuestionOLX[0].problem;
+      this.questionData = this.richTextOLX[0].problem;
+      this.richTextProblem = this.richTextOLX[0].problem;
+    }
+  }
+
+  /** getPreservedAnswersAndFeedback(problemType, widgetName, option)
+   * getPreservedAnswersAndFeedback takes a problemType, widgetName, and a valid option. The
+   * olx for the given problem type and widget is parsed. Depending on the
+   * @param {string} problemType - string of the olx problem type
+   * @param {string} widgetName - string of the wrapping tag name
+   *                              (optioninput, choicegroup, checkboxgroup, addition_answer)
+   * @param {string} option - string of the type of answers (choice, option)
+   * @return {array} array containing answer objects and possibly an array of grouped feedback
+   */
+  getPreservedAnswersAndFeedback(problemType, widgetName, option) {
+    const preservedProblem = this.richTextProblem;
+    const isChoiceProblem = problemType !== ProblemTypeKeys.NUMERIC ? problemType !== ProblemTypeKeys.TEXTINPUT : false;
```

Comment:
This is hard to understand. I needed a while to work through all the negations to figure out what value is actually returned. I played around a bit, this seems a more readable version I came up with:
```
const isChoiceProblem = !([ProblemTypeKeys.NUMERIC, ProblemTypeKeys.TEXTINPUT].includes(problemType))
```